junit-resource-poller/src/main/java/com/palantir/junit/PollableResource.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.junit;
+
+
+import com.google.common.base.Optional;
+
+public interface PollableResource {
+    /**
+     * Returns {@link Optional#absent} iff the the resource is available, or an exception explaining why the resource is
+     * not considered ready else.
+     */
+    Optional<Exception> isReady();
```

Comment:
there's really nothing wrong with this, but it seems strange that the positive result of this method is an `Optional.absent()`